lib/debootstrap.sh

```diff
@@ -668,13 +676,11 @@ PREPARE_IMAGE_SIZE
 		fi
 
 		check_loop_device "$rootdevice"
-		display_alert "Creating rootfs" "$ROOTFS_TYPE on $rootdevice"
-		mkfs.${mkfs[$ROOTFS_TYPE]} ${mkopts[$ROOTFS_TYPE]} ${mkopts_label[$ROOTFS_TYPE]:+${mkopts_label[$ROOTFS_TYPE]}"$ROOT_FS_LABEL"} $rootdevice >> "${DEST}"/${LOG_SUBPATH}/install.log 2>&1
+		display_alert "Creating root fs" "${ROOTFS_TYPE} on ${rootdevice}"
+		mkfs.${mkfs[$ROOTFS_TYPE]} ${mkopts[$ROOTFS_TYPE]=:''} ${mkopts_label[$ROOTFS_TYPE]:+${mkopts_label[$ROOTFS_TYPE]}${ROOT_FS_LABEL}} ${rootdevice} >> ${DEST}/${LOG_SUBPATH}/install.log 2>&1
 		[[ $ROOTFS_TYPE == ext4 ]] && tune2fs -o journal_data_writeback $rootdevice > /dev/null
-		if [[ $ROOTFS_TYPE == btrfs && $BTRFS_COMPRESSION != none ]]; then
-			local fscreateopt="-o compress-force=${BTRFS_COMPRESSION}"
-		fi
-		mount ${fscreateopt} $rootdevice $MOUNT/
+		[[ $ROOTFS_TYPE == btrfs && $BTRFS_COMPRESSION != none ]] && local rfsmntopt='-o compress-force='${BTRFS_COMPRESSION}
+		mount ${rfsmntopt:=} ${rootdevice} ${MOUNT}
```

Comment:
Necessity? bash will parse undefined variable as empty string.

Reply:
Indeed thought so initially. But introduced expansion with well defined unset/null behaviour, because when i tried the code in a simplified form but actually making a partitioned disk image, I found that as part of the command line to mkfs.fstype the unset parameter did cause problems.

Reply:
Can you explain more? How to reproduce? If undefined variable isn't be treated as empty string. The script may have more problem.